sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -277,6 +282,74 @@ private Mono<ConfidentialClientApplication> getConfidentialClientApplication() {
         });
     }
 
+    private Mono<ConfidentialClientApplication> getManagedIdentityConfidentialClient() {
+        return Mono.defer(() -> {
+            String authorityUrl = TRAILING_FORWARD_SLASHES.matcher(options.getAuthorityHost()).replaceAll("")
+                + "/" + tenantId;
+
+            // Temporarily pass in Dummy Client secret and Client ID. until MSal removes its requirements.
+            IClientCredential credential = ClientCredentialFactory.createFromSecret("dummy-secret");
+            ConfidentialClientApplication.Builder applicationBuilder =
+                ConfidentialClientApplication.builder(clientId == null ? IdentityConstants.DEVELOPER_SINGLE_SIGN_ON_ID
+                    : clientId, credential);
+            try {
+                applicationBuilder = applicationBuilder.authority(authorityUrl);
+            } catch (MalformedURLException e) {
+                return Mono.error(LOGGER.logExceptionAsWarning(new IllegalStateException(e)));
+            }
+
+            applicationBuilder.appTokenProvider(appTokenProviderParameters -> {
+                TokenRequestContext trc = new TokenRequestContext()
+                    .setScopes(new ArrayList<>(appTokenProviderParameters.scopes))
+                    .setClaims(appTokenProviderParameters.claims)
+                    .setTenantId(appTokenProviderParameters.tenantId);
+
+                Mono<AccessToken> accessTokenMono = getTokenFromTargetManagedIdentity(options.getManagedIdentityType(), trc);
```

Comment:
nit: appending Mono to the name doesn't seem to be our pattern.
```suggestion
                Mono<AccessToken> accessToken = getTokenFromTargetManagedIdentity(options.getManagedIdentityType(), trc);
```